tests/test_product_pages_api.py

```diff
@@ -0,0 +1,180 @@
+# SPDX-License-Identifier: GPL-3.0+
+import requests_mock
+
+from retasc.product_pages_api import ProductPagesApi
+
+PP_URL = "https://product_pages.example.com"
+PP_API = ProductPagesApi(PP_URL)
+
+
+def test_get_pp_url_info():
+    assert PP_API.api_url == PP_URL
+    assert PP_API.api_url_releases() == PP_URL + "/releases/"
+
+
+def test_releases_by_product_phase():
+    default_fields = "shortname,product_shortname,ga_date,phase_display"
+    test_prod = "rhscl"
+    res = [
+        {
+            "shortname": "rhscl-3-8",
+            "product_shortname": test_prod,
+            "ga_date": "2021-11-15",
+            "phase_display": "Maintenance",
+        }
+    ]
+
+    # test all default parameters for ProductPagesApi.releases_by_product_phase, url should has all default fields
+    with requests_mock.Mocker() as m:
+        m.get(
+            f"{PP_API.api_url_releases()}?product__shortname={test_prod}&fields={default_fields}&phase__lt=Unsupported",
```

Comment:
This could be difficult to maintain. It would be better to have a helper test function to help with testing the parameters (maybe with `additional_matcher`: https://requests-mock.readthedocs.io/en/latest/matching.html#additional-matchers). Or check later if expected API endpoint was called with specific parameters (https://requests-mock.readthedocs.io/en/latest/history.html):

```python
    with requests_mock.Mocker() as m:
        m.get(PP_API.api_url_releases(), json=res)
        resp = PP_API.releases_by_product_phase(test_prod)
        assert len(m.request_history) == 1
        req = m.request_history[0]
        assert ("product__shortname", test_prod) in req.params
        # these are probably not important to test - probably ok to drop
        assert ("phase__lt", "Unsupported") in req.params
        assert ("fields", default_fields) in req.params
```